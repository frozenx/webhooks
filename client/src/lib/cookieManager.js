export default {
	readCookie(key) {
		let cookieValue = ''
		const cookieArray = document.cookie.split(';')
		cookieArray.forEach((cookieString) => {
			const cookiePair = cookieString.trim().split('=')
			if (cookiePair[0] === key) {
				cookieValue = cookiePair[1]
			}
		})
		return cookieValue
	},
	setCookie(key, value, domain = '.tesco.com') {
		document.cookie = `${key}=${value};path=/;domain=${domain}`
	},
}
