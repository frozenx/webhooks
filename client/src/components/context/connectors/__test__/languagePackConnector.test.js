jest.mock('../../../../lib/cookieManager');

import React from 'react';
import renderer from 'react-test-renderer';
import LanguagePackConnector from '../languagePackConnector';
import cookieManager from '../../../../lib/cookieManager';


describe('languagePackConnector', () => {
    beforeEach(() => {
        cookieManager.readCookie.mockReturnValue('en');
        // global.navigator = { languages: ['en'], language: 'en' };
    });
    afterEach(() => {
        cookieManager.readCookie.mockRestore();
    });
    test('should render the component along with its children given cookie is set', () => {
        const tree = renderer.create(<LanguagePackConnector />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('should render component taking language from navigator given cookie is not set', () => {
        cookieManager.readCookie.mockReturnValue(null);
        const tree = renderer.create(<LanguagePackConnector />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});