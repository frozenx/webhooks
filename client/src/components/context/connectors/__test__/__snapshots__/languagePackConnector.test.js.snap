// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languagePackConnector should render component taking language from navigator given cookie is not set 1`] = `null`;

exports[`languagePackConnector should render the component along with its children given cookie is set 1`] = `null`;
